### ISSUE TITLE
InfluxDB Client /query endpoint: invalid character 'M' looking for beginning of value

### ISSUE THREAD

**Issue by subfuzion** [influxdata/influxdb#7184]
### Bug report

This is the same problem that was observed in #6540, implying a version mismatch between the client and the server:

`unable to decode json: received status code 405 err: invalid character 'M' looking for beginning of value`

However, what makes this a head-scratcher is that it's working under Docker on Linux but not OS X with the same images (and double-checking that the tagged client package matches the server version). I confirmed the error for both `0.13` and `v1.0.0-beta3`.

Obviously that makes it sound like maybe an issue with Docker for Mac, and yet we haven't had any other issues in an environment that includes etcd, kafka, and elasticsearch.

Has anyone else encountered this or have any ideas on resolving the issue?

**Comment by jsternberg** [comment]
I wonder if you have a local copy running on your Mac. How are you connecting to the Docker container on Mac OS X? On Linux, Docker typically runs on localhost and you're fine. With Mac OS X, you're using a Virtual Machine and need to use a different IP address.

**Comment by subfuzion** [comment]
We tested on a brand new pristine Mac, same issue.

**Comment by jsternberg** [comment]
What does the InfluxDB log say after you make a query? Does it record a `GET` or `POST`?

**Comment by subfuzion** [comment]
Similar to the referenced issue, https://github.com/influxdata/influxdb/issues/6540#issuecomment-216408832, it doesn't get that far.

**Comment by subfuzion** [comment]
I forgot to mention in the description, but what also makes this a head-scratcher is that curl requests do succeed.

**Comment by jsternberg** [comment]
Can you show me the curl command that succeeds?

**Comment by subfuzion** [comment]
Simple example, but they all succeed:

```
curl -X GET 'http://127.0.0.1:8086/query?q=SHOW+MEASUREMENTS&db=_internal'
```

**Comment by jsternberg** [comment]
Can you try that same query with a POST? If it doesn't work, then your
local copy isn't up to date. As mentioned earlier, you shouldn't be using
localhost if you're using Docker on a Mac so there might be a local copy
you don't know about running an old version.

**Comment by subfuzion** [comment]
Hi @jsternberg, thanks for the follow up. I don't have influxdb installed on the system and also tested on a pristine Mac with only Docker installed. In development, our machines are swarm managers and localhost with Docker for Mac is appropriate for connecting to local swarm services (started via `docker service create...`).

I did try the query with a POST and it failed. This was a great suggestion on your part and the clue I needed. After investigation, it appears that the latest image on our registry had been inadvertently updated while one of the team members was experimenting with different versions. It hadn't affected anyone else who hadn't pulled the image as recently as me. It was coincidental that I happened to pull on two different machines running Docker for Mac. The issue is resolved (and we're updating the script to make sure it pulls the specific desired version for the client, not `latest`). Thank you very much for your help.
